app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -182,6 +184,7 @@ public class MainActivity : AppCompatActivity(), MainViewController, RouteCallba
         app?.onActivityResume()
         autoCompleteAdapter?.clear()
         autoCompleteAdapter?.notifyDataSetChanged()
+        invalidateOptionsMenu()
```

Comment:
Is this really required here? According to the docs for `onPrepareOptionsMenu()` it is called `called right before the menu is shown, every time it is shown.` Of course that doesn't mean this is the actual behavior ;)

https://developer.android.com/reference/android/app/Activity.html#onPrepareOptionsMenu(android.view.Menu)

On the other hand if `invalidateOptionsMenu()` is required to update whether we are caching search results I would think this needs to be called from `onResume()` since the activity is not destroyed but simply goes into the background when the settings activity is started.